scripts/bee.sh

```diff
@@ -25,30 +25,50 @@ USAGE
     exit 1
 }
 
-stop() {
+stop_containers() {
     echo "Stop Bee following containers:"
     docker container stop "$QUEEN_CONTAINER_NAME";
     WORKER_NAMES=$(docker container ls -f name="$WORKER_CONTAINER_NAME*" --format "{{.Names}}")
     for WORKER_NAME in $WORKER_NAMES; do
         docker container stop "$WORKER_NAME"
     done
+}
 
+stop() {
+    stop_containers
     trap - SIGINT
     exit 0;
 }
 
+queen_failure() {
+    docker logs "$QUEEN_CONTAINER_NAME"
+    stop_containers
+    echo "Timeout limit has been reached, exit from the process.."
+    exit 1
+}
+
 fetch_queen_underlay_addr() {
     if [[ -n "$QUEEN_UNDERLAY_ADDRESS" ]] ; then return; fi
 
-    while : ; do
-        QUEEN_UNDERLAY_ADDRESS=$(curl -s localhost:1635/addresses | python -mjson.tool 2>&1 | grep "/ip4/" | awk '!/127.0.0.1/' | sed 's/,$//' | xargs)
+    ELAPSED_TIME=0
+    WAITING_TIME=5
+    # Wait 2 mins for queen start
+    TIMEOUT=$((2*12*WAITING_TIME))
```

Comment:
I find this a bit strange, IMHO just put in 120. Because if someone alters the `WAITING_TIME` this also gets changes and at minimum the comment `# Wait 2 mins` is no longer true